backend/db/crud_data.py

```diff
@@ -91,8 +92,22 @@ def get_data(
     return PaginatedData(data=data, count=count)
 
 
-def get_all_data(session: Session, registration: bool) -> DataDict:
-    return session.query(Data).filter(Data.registration == registration).all()
+def get_all_data(
+    session: Session,
+    registration: bool,
+    options: List[str] = None,
+) -> DataDict:
+    data = session.query(Data).filter(Data.registration == registration)
+    if options:
+        # support multiple select options filter
+        # change query to filter data by or_ condition
+        or_query = or_(
+            ViewAdvanceFilter.options.contains([opt]) for opt in options
+        )
+        data_id = session.query(ViewAdvanceFilter.data).filter(or_query).all()
```

Comment:
Here also we need to change the query after fixing `advance_filter` materialized view.
The query should get the `identifier` value:
`data_id = session.query(ViewAdvanceFilter.identifier).filter(or_query).all()`